### PR TITLE
Implement API endpoints for managing Twilio TaskRouter resources

### PR DESCRIPTION
**Pull request by qu8n**
Adding Next.js API endpoints for managing the following Twilio TaskRouter resources:
- Workspaces
- Workflows
- Workers
- TasksQueues
- Tasks
- Activities

Also included is a frontend UI for performing CRUD operations on Workspaces. This is a placeholder UI only for demo purposes and not meant to be part of the actual UI implementation.

Other notes:
- See the README I added for starting the dev server with the required Twilio credentials. Anyone can open a free trial account [here](https://www.twilio.com/try-twilio) with ~$15 in credits.
- The UI was built using TailwindUI. Because we're going with Material UI (if I recall correctly), we should remove the TailwindUI resources in the future. This includes removing packages `@tailwindcss/forms` and `@heroicons/react`, as well as removing the `@tailwindcss/forms` plugin inside `tailwind.config.js`.
- The standalone Next.js app with these API endpoints can be found in [this private repo](https://github.com/qu8n/twilio-taskrouter-wrapper). Let me know if anyone wants to be added to test it individually.

### CODE REVIEW

File: src/pages/workspaces/[sid].tsx [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+  }, [callbackEvents])
+
+  useEffect(() => {
+    setWorkspace({
+      ...workspace,
+      multiTaskEnabled,
+    })
+  }, [multiTaskEnabled])
+
+  // console.log(activities)
```
Comment (ZzRanger) [inline src/pages/workspaces/[sid].tsx]: Nit: remove unused code

File: src/pages/workspaces/[sid].tsx [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+  const [multiTaskEnabled, setMultiTaskEnabled] = useState<boolean>(
+    currWorkspace.multiTaskEnabled
+  )
+  const [allEvents, setAllEvents] = useState<boolean>(
+    currWorkspace.eventsFilter === ''
+  )
+  const [selectAll, setSelectAll] = useState<boolean>(
+    currWorkspace.eventsFilter === ''
+  )
+  const [callbackEvents, setCallbackEvents] = useState<
```
Comment (ZzRanger) [inline src/pages/workspaces/[sid].tsx:26]: You're a champ for this 😤

File: src/lib/callbackEventsData.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
+export const callbackEventsData: Record<
```
Comment (ZzRanger) [inline src/lib/callbackEventsData.tsx:1]: Also for this frfr

File: src/pages/workspaces/[sid].tsx [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+          (
+            acc: Record<
+              string,
+              { description: string; eventId: string; enabled: boolean }
+            >,
+            [key, value]
+          ) => {
+            acc[key] = {
+              ...value,
+              enabled: currWorkspace.eventsFilter
```
Comment (ZzRanger) [inline src/pages/workspaces/[sid].tsx:41]: Add a check here cause sometimes eventsFilter will sometimes be null.

Screenshot 2023-06-17 at 3 59 18 PM

**Review by ZzRanger — APPROVED** [review]
Great work overall, this is amazing and will give developers on a team a great place to start interacting with the Twilio APIs! Just resolve the comments then we'll merge :)

File: src/pages/workspaces/[sid].tsx (hunk 4 above)
Comment (qu8n) [reply src/pages/workspaces/[sid].tsx:41]: Good callout. Thank you.

Fixed. See line 29 of the latest code in `src/pages/workspaces/[sid].tsx`. I'm using the ternary operator to handle when eventsFilter is null or an empty string (i.e. falsy).

**Review by ZzRanger — APPROVED** [review]
LGTM! Merge it whenever

**Pull request merged**